sklearn/utils/estimator_checks.py

```diff
@@ -2440,7 +2443,10 @@ def check_classifiers_train(
                 # decision_function agrees with predict
                 decision = classifier.decision_function(X)
                 if n_classes == 2:
-                    if not tags["multioutput_only"]:
+                    if (
+                        not tags.target_tags.multi_output
+                        or tags.target_tags.single_output
```

Comment:
Shouldn't this be 
```suggestion
                        tags.target_tags.multi_output
                        or tags.target_tags.single_output
```
as the previous condition states `not multioutput_only` which seems to suggest it can be multioutput or single output?

Reply:
It's confusing, but writing the logic then you get: 

```py
not multioutput_only
==
not (multi_output and not single_output)
==
not multi_output or single_output
```